src/index.js

```diff
@@ -1533,6 +1582,50 @@ function loadSong( n ) {
 		return false;
 }
 
+/**
+ * Copy the gradient of given key into currentGradient, and render the gradient editor.
+ */
+function openGradientEdit(key) {
+	loadGradientIntoCurrentGradient(key);
+	renderGradientEditor();
+	$('#btn-save-gradient').innerText = 'Save';
+	$('#btn-delete-gradient').style.display = 'block';
+	location.href = '/#gradient-editor';
+}
+
+/**
+ * Build a new gradient, set it as the current gradient, then render the gradient editor.
+ */
+function openGradientEditorNew() {
+	currentGradient = {
+		dir: 'v',
```

Comment:
`'v'` has no actual function here, as the analyzer module ignores any value other than `'h'` for `dir`. For consistency with the other gradient definitions, I think we may simply leave `dir` out of the initial object. In the future, I may improve this in the module by changing it to a boolean property.